app/renderer/components/tabs/tab.js

```diff
@@ -153,19 +156,26 @@ class Tab extends React.Component {
   onMouseLeave (e) {
     // mouseleave will keep the previewMode
     // as long as the related target is another tab
+    clearTimeout(this.mouseTimeout)
     windowActions.setTabHoverState(this.props.frameKey, false, hasTabAsRelatedTarget(e))
   }
 
   onMouseEnter (e) {
     // if mouse entered a tab we only trigger a new preview
     // if user is in previewMode, which is defined by mouse move
+    clearTimeout(this.mouseTimeout)
     windowActions.setTabHoverState(this.props.frameKey, true, this.props.previewMode)
   }
 
   onMouseMove () {
     // dispatch a message to the store so it can delay
     // and preview the tab based on mouse idle time
-    windowActions.onTabMouseMove(this.props.frameKey)
+    clearTimeout(this.mouseTimeout)
+    this.mouseTimeout = setTimeout(
+      () => {
+        windowActions.setTabHoverState(this.props.frameKey, true, true)
```

Comment:
no matter what we do with this in terms of the timeout, I don't think we should go back to an imperative action. onmouseenter/leave should be changed at some point as well

Reply:
agreed- should have a name like windowActions.onTabHoverChanged

Reply:
Because it would take some work to either a) create an action like this or b) modify the existing uses of the action which could use a better name, I think that work should be deferred until the code behind setTabHoverState is touched.  Trying to capture this use case and the existing use-cases with one name would be tricky. The long term solution would be to move this logic to reducers